src/geo/transform.js

```diff
@@ -1306,6 +1310,48 @@ class Transform {
         return this._mercatorZfromZoom(this._minZoomForBounds());
     }
 
+    _calcFogMatrices() {
+        if (!this.fogEnd || !this.fogCulling) {
+            return;
+        }
+
+        const maxVisibleAltitude = this.elevation ? this.elevation.visibleDemTiles.reduce((res, tile) => {
+            if (tile.dem) {
+                const tree = tile.dem.tree;
+                const maxMip = tree.maximums[0];
+                res = Math.max(res, maxMip);
+            }
+            return res;
+        }, 0) * this.elevation.exaggeration() : 0;
+
+        const projMaxVisibleAltitude = (maxVisibleAltitude * this.pixelsPerMeter) / Math.cos(this._pitch);
+        const fogEnd = this.fogEnd + projMaxVisibleAltitude;
+        const camZ = this._camera.position[2] * this.worldSize;
+        const groundAngleAtFogEnd = Math.atan(camZ / fogEnd);
+        const camToFogEnd = Math.hypot(camZ, fogEnd);
+        const farFogEnd = Math.sin(Math.PI - groundAngleAtFogEnd - this._pitch) * camToFogEnd;
+
+        const cameraToClip = this._camera.getCameraToClipPerspective(this._fov, this.width / this.height, this._nearZ(), farFogEnd);
+
+        cameraToClip[8] = -this.centerOffset.x * 2 / this.width;
```

Comment:
Is it possible to label the components/meaning when accessing matrix elements directly? This looks like maybe the x/y entries of the third column of a matrix, but it's not particularly clear.

Reply:
Ended up removing this section for more fine-grained culling in the tile cover (e987ba4 and 0555663).